core/ledger/ledger_interface.go

```diff
@@ -215,6 +215,13 @@ type PeerLedger interface {
 	CancelSnapshotRequest(height uint64) error
 	// PendingSnapshotRequests returns a list of heights for the pending (or under processing) snapshot requests.
 	PendingSnapshotRequests() ([]uint64, error)
+	// CommitNotificationsChannel returns a read-only channel on which ledger sends a `CommitNotification`
+	// when a block is committed. The CommitNotification contains entries for the transactions from the committed block,
+	// which are not malformed, carry a legitimate TxID, and in addition, are not marked as a duplicate transaction.
+	// The consumer can close the 'done' channel to signal that the notifications are no longer needed. This will cause the
+	// CommitNotifications channel to close. There is expected to be only one consumer at a time. The function returns error
+	// if already a CommitNotification channel is active.
+	CommitNotificationsChannel(done <-chan struct{}) (<-chan *CommitNotification, error)
```

Comment:
A better approach than the existing state listeners.

Reply:
I don't think that the both are comparable. The existing listeners (e.g., to create couchdb index, update confighistory, and update lifecycle cache are mainly designed to process as part of the transaction. E.g., if they fail, transaction fails. So, they are inline processing. In this case, it's a producer-consumer model, so channel fits naturally.

Reply:
Agree. I raised this point because as soon as I read the commit notification I was wondering whether we can reuse the state listener. However, I understood that it would not fit here. Further, the channel approach looked better than the callback. 

Yes `HandleStateUpdates()` returns an error but it is not required in this notification channel.